### ISSUE TITLE
More intro material from a Rustacean point of view

### ISSUE THREAD

**Issue by Manishearth** [unicode-org/icu4x#2675]
I think ICU4X's readme and tutorials are largely written from the point of view of "I am an internationalization user and I want to use ICU4X", which is quite reasonable since that is the majority of our clients.

I do think that we should also try and pay attention to the potential user base of Rustaceans who may not necessarily know much about internationalization.

I do try to make some fixes along these lines in https://github.com/unicode-org/icu4x/pull/2674 (inspired by feedback from @Gankra) but we probably need deeper changes.

A nice thing about the Rust community is that in general folks tend to be rather curious and interested in other Rust projects, even if they don't plan to use it any time soon. This leads to a lot of foot traffic on libraries due to curiousity rather than need, and I think so far  ICU4X's tutorials and readme (and the draft 1.0 blog post) target people coming here from _need_. It would be good for ICU4X to also target people arriving here out of curiosity, which will be great for growing the group of people who will wish to advertise the library when they hear someone has needs that fit, and also leading to potential contributors. I've contributed to _plenty_ of Rust libraries that I have basically no intention of using, because the libraries were cool.

I think I see two questions we could be answering better:

 - Why should I, a Rustacean, care about this library? Why would I want to use this?
 - I'm a Rustacean wanting to internationalize my application, what should I know?

The latter one is technically broader in scope than just ICU4X (people wanting to internationalize will likely want to use something like fluent as well, and also some translation management interface, and a bunch of other such things), but we can try to answer that question in the context of ICU4X at least, to make it easier for whoever eventually writes a general guide of this form.

cc @sffc @zbraniecki @echeran

**Comment by zbraniecki** [comment]
I agree. I think we need to continue building on recent improvements - I think our API docs are very good (not great yet!), and we have tutorials that provide the onboarding *when necessary*.

What I'd like us to improve is to go one step further in two directions:
* Introduce i18n. I understand we won't have a lot of time for it, but bridging the gap from "I stumbled upon ICU4X and it tells me it solves internationalization. What is internationalization? What makes it complex? What are major concepts?" would be great.
* Have a smooth rabbit hole dives from "Here is your first use of ICU4X" to "i18n date is hard, integration with app is hard, we developed a very very sophisticated system to handle it, and you will need to think how you want to use it". Shane wrote this in APi docs, now linking it from tutorials is going to help.

**Comment by Manishearth** [comment]
Yeah that's a good sketch I think!

**Comment by echeran** [comment]
> * Introduce i18n. I understand we won't have a lot of time for it, but bridging the gap from "I stumbled upon ICU4X and it tells me it solves internationalization. What is internationalization? What makes it complex? What are major concepts?" would be great.

The [videos from Unicode's Overview of Internationalization and Unicode Projects](https://www.youtube.com/playlist?list=PLMc927ywQmTNQrscw7yvaJbAbMJDIjeBh) event last week are now live on the interwebs.  They should do a pretty good job of surveying high level concepts about i18n.  The videos even include [the presentation about ICU4X](https://youtu.be/-sICg_Y3N8s), which announces the 1.0 release, by our very own @sffc !

**Comment by Manishearth** [comment]
Yeah, however I don't think video content is ideal for our main entry point and I think having stuff focused towards potential icu4x users would be good
